api/helpers/users.js

```diff
@@ -2,9 +2,13 @@ const models = require('../../database/models');
 
 module.exports = {
   async findUserByUsername(username) {
-    const user = await models.Users.findOne({
-      where: { username }
-    });
-    return user;
+    try {
```

Comment:
How do we know when this function returns an error or a valid data?

Reply:
check the catch block the error.message is returned in there

Reply:
Yeah, shouldn't the method the calls it be the one to catch the error and handle it, according to it's need. Because this way, it doesn't ever throw an error when it fails

Reply:
since it's an asynchronous function when it's called wherever the error will move to the catch block also as the promise is either resolved or rejected.
And yes it returns the corresponding error